pxd.c

```diff
@@ -1347,6 +1349,21 @@ ssize_t pxd_add(struct fuse_conn *fc, struct pxd_add_ext_out *add)
 		return pxd_dev->minor;
 	}
 
+	/* pre-check to detect if prior instance is removed */
+	sprintf(devfile, "/dev/pxd/pxd!pxd%llu", add->dev_id);
+	err = vfs_stat(devfile, &pxdev_stat);
+	if (err == 0) {
+		pr_err("stale device(%s) found, attach fail", devfile);
+		return -EEXIST;
```

Comment:
goto appropriate cleanup label instead of return

Reply:
revised.